curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/65420dcfc60580678dad7a92.md

```diff
@@ -0,0 +1,619 @@
+---
+id: 65420dcfc60580678dad7a92
+title: Step 13
+challengeType: 0
+dashedName: step-13
+---
+
+# --description--
+
+Inside the `map()`, add a `return` statement with backticks where you will interpolate all the elements responsible to displaying the song details.
+
+Inside the backticks, create an `li` tag with the class `playlist-song`. Add an `id` attribute with the value of `song-${song.id}` to the `li` tag.
+
+# --hints--
+
+Test 1
+
+```js
+
+```
```

Comment:
The tests for this step:

You should add a `return` statement with an empty pair of backticks.

```js
assert.match(code, /return\s*`/)
```

You should create an `li` tag inside the backticks.

```js
assert.match(code, /return\s*`\s*<li\s*/)
```

Your `li` tag should have the `id` `song-${song.id}` as its first attribute and value.

```js
assert.match(code, /return\s*`\s*<li\s*id=('|")song-\$\{song\.id\}\1\s*/)
```

Your `li` tag should have the class `playlist-song` as its second attribute and value.

```js
assert.match(code, /return\s*`\s*<li\s*id=('|")song-\$\{song\.id\}\1\s*class=\1playlist-song\1>\s*<\/li>\s*`;?/)
```